src/main/java/com/twcable/grabbit/tools/cli/GrabbitCli.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2014-2016 Time Warner Cable, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.twcable.grabbit.tools.cli;
+
+import com.twcable.grabbit.tools.environment.Environment;
+import com.twcable.grabbit.tools.environment.HostInfo;
+import com.twcable.grabbit.tools.environment.MapSourcedEnvironment;
+import com.twcable.grabbit.tools.jobstarter.HostAndJobId;
+import com.twcable.grabbit.tools.jobstarter.JobStarter;
+import com.twcable.grabbit.tools.jobstarter.JobsConfigFileReader;
+import com.twcable.grabbit.tools.monitor.JobStatusCache;
+import com.twcable.grabbit.tools.monitor.PollingJobMonitor;
+import com.twcable.grabbit.tools.monitor.SimpleFileJobStatusCache;
+import com.twcable.grabbit.tools.monitor.event.MonitoringEvent;
+import com.twcable.grabbit.tools.util.Either;
+import com.twcable.grabbit.tools.util.Utils;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.WorkQueueProcessor;
+import reactor.core.subscriber.Subscribers;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.stream.Collectors;
+
+import static com.twcable.grabbit.tools.monitor.JobState.RUNNING;
+
+/**
+ * The command line entry point.
+ */
+@SuppressWarnings({"WeakerAccess", "RedundantTypeArguments"})
+public class GrabbitCli {
+    public static final String DEFAULT_JOB_STATUS_CACHE_FILENAME = "grabbitIds.out";
+
+
+    @SuppressWarnings({"PMD.UseVarargs", "PMD.SystemPrintln"})
+    public static void main(String[] args) throws IOException, InterruptedException {
+        final Either<String, CliOptions> cliOptions = CliOptions.create(args);
+
+        if (cliOptions.isRight()) {
```

Comment:
Just curious, why not subtype something specific from CliOptions? CliValidOptions/CliBadOption

Is it because the types seem a bit disjoint in this example? That, or it's almost like a way to return a tuple...which is kinda cool